docker/entrypoint.py

```diff
@@ -6,28 +6,83 @@
+    """Extract a file to a destination directory.
+
+    Args:
+        source (Path): Source directory to extract from
+        destination (Path): Destination directory to extract to
+        prefix (str, optional): Prefix extracted filepaths. Default is "/".
+        kind (str, optional): File extension to extract. Default is "war".
+
+    Raises:
+        FileNotFoundError: If source directory does not exist.
     """
-    if not os.listdir(webapps_dir):
-        for war in os.listdir(webapps_ref):
-            if war.endswith(".war"):
-                fn = os.path.splitext(war)[0]
-                prefix = path_prefix.replace("/", "#").strip("#")
-                war_dir = os.path.join(webapps_dir, f"{prefix}#{fn}" if prefix else fn)
-                os.makedirs(war_dir, exist_ok=True)
-                subprocess.call(["unzip", "-oqd", war_dir, os.path.join(webapps_ref, war)])
-                subprocess.call(["sed", "-E", "-i.bak", "s/##out--//", os.path.join(war_dir, "WEB-INF/classes/log4j2.properties")])
+    extractions: dict[str, Path] = {}
+    print(f"{SLIDES} Extracting Files {SLIDES}")
+    print(f"Source     : {source}")
+    print(f"Destination: {destination}")
+    print(f"Prefix     : {prefix}")
+    print(f"Kind       : {kind}")
+    print("\n")
+
+    if not source.exists() or not source.is_dir():
+        raise FileNotFoundError(f"Source directory does not exist: {source}")
+
+    for item in source.iterdir():
+        # Only extract specified file extensions
+        if item.is_file() and item.suffix == f".{kind}":
+            filename = item.stem
+            print(f"Extracting: {item}")
+            # Configure prefix for extracted files
+            prefix = prefix.replace("/", "#").strip("#")
+            if prefix:
+                landing = destination / f"{prefix}#{filename}"
+            else:
+                landing = destination / filename
+
+            # Create landing directory if it does not exist
+            print(f"    Preparing -> {landing}")
+            landing.mkdir(parents=True, exist_ok=True)
+            # Extract archive to landing directory
+            print(f"    Extracting {item} -> {landing}")
+            with ZipFile(item, "r") as archive:
+                archive.extractall(landing)
+            extractions[filename] = landing
+            # Modify log4j to send logs to stdout
+            log4j = destination / "WEB-INF/classes/log4j2.properties"
+            if log4j.exists():
+                # Create backup as log4j.properties.bak
+                print(f"    Modifying {log4j}")
+                shutil.copy(log4j, log4j.with_suffix(".bak"))
+                with open(log4j, "r+", encoding="utf-8") as file:
+                    content = file.read()
+                    content = content.replace("##out--", "")
+                    file.seek(0)
+                    file.write(content)
+                    file.truncate()
```

Comment:
This doesn't seem to be working. The file remains unmodified, and no logs are being sent to the console.

Reply:
I will get this fixed.

Reply:
The issue has been fixed, I believe since I was overloading the files at `destination / "WEB-INF/classes/log4j2.properties"` rather than `landing / "WEB-INF/classes/log4j2.properties"`.